### PR TITLE
Rename `ProtocolDisplay` to `ProtocolBridge`

### PR DESCRIPTION
**Pull request by JakeWharton**
We are going to have a symmetric type on the Compose side soon which will share its name.

Going for
```mermaid
flowchart LR;
    Compose-->DiffProducingWidgetFactory;
    DiffProducingWidgetFactory-->PB1[ProtocolBridge];
    PB1[ProtocolBridge] --> Protocol;
    Protocol --> ProtocolBridge;
    ProtocolBridge -->DiffConsumingWidgetFactory;
    DiffConsumingWidgetFactory-->Widgets;

```

### CODE REVIEW

**Review by colinrtwhite — APPROVED** [review]
Just to make sure I understand - so the plan is to eventually be able to pass a custom `Protocol` to `ProtocolRedwoodComposition` so we can skip diff serialization/deserialization if we're not using Zipline?

File: redwood-protocol-widget/src/commonMain/kotlin/app/cash/redwood/protocol/widget/ProtocolBridge.kt [hunk 1]
```diff
@@ -22,17 +22,25 @@ import app.cash.redwood.protocol.Diff
...
 
-public class ProtocolDisplay<T : Any>(
+/**
+ * Bridges the serialized Redwood protocol back to widgets on the display side.
+ *
+ * This type will consume [Diff]s and apply their [ChildrenDiff] operations to the widget tree.
+ * [PropertyDiff]s and [LayoutModifiers]s are forwarded to their respective widgets. Events from
+ * widgets are forwarded to [eventSink].
+ */
+public class ProtocolBridge<T : Any>(
```
Comment (colinrtwhite) [inline redwood-protocol-widget/src/commonMain/kotlin/app/cash/redwood/protocol/widget/ProtocolBridge.kt:37]: Why have two types with the same name if one takes diffs as inputs and the other as outputs?

**Comment by JakeWharton** [comment]
If you don't want to use the protocol to serialize the values then the Compose code will talk directly to the Widget code. It already does this today, as the diff-producing factory implements the same widget factory that you would implement. So you choose whether it talks directly to your widget factory or whether it talks to a widget factory that records operations as diffs in the protocol bridge. Then, on the other side, there's an inverse protocol bridge that unwraps diffs, talks to a diff-consuming widget factory, and then it talks to your real widget factory.

File: redwood-protocol-widget/src/commonMain/kotlin/app/cash/redwood/protocol/widget/ProtocolBridge.kt (hunk 1 above)
Comment (JakeWharton) [reply redwood-protocol-widget/src/commonMain/kotlin/app/cash/redwood/protocol/widget/ProtocolBridge.kt:37]: I couldn't think of a good-enough descriptive name. They each have inputs and outputs–they're just mirrors of each other. The compose-side has diff outputs and event inputs and the widget-side has event outputs and diff inputs.

**Pull request merged**